core-bundle/src/Security/Voter/BackendAccessVoter.php

```diff
@@ -129,6 +129,29 @@ private function hasAccess(mixed $subject, string $field, BackendUser $user): bo
             return !empty($this->pagemountsCache[$user->id]) && array_intersect($subject, $this->pagemountsCache[$user->id]);
         }
 
+        // Additionally check for 'disablePermissionChecks' flag for modules
+        if ('modules' === $field) {
+            $hasModuleAccess = null;
+
+            foreach ($subject as $module) {
+                foreach ($GLOBALS['BE_MOD'] as $modules) {
+                    foreach ($modules as $name => $config) {
+                        if ($name === $module) {
+                            $hasModuleAccess = $config['disablePermissionChecks'] ?? false;
+                        }
+
+                        if (false === $hasModuleAccess) {
+                            break 3;
```

Comment:
Wouldn't it be easier to just use an early return here?

Reply:
Check is now simplified 👍

Reply:
If there are multiple checks they should probably each go in a separate function, so you can easily grasp it at the higher level/main function. 👍